### PR TITLE
fix: parameter aggregation fix

### LINKED ISSUE #1442: Parameter aggregation failure
## Describe the bug

Parameter aggregation doesn´t work accordingly. A parameter aggregation is recognized when a parameter `a` contains in it´s props a prop highlighting that the parameter aggregates its value from another parameters `b & c`. e.g

```
"params": [
              {
                "id": "a",
                "props": [
                  {
                    "name": "aggregates",
                    "ns": "http://csrc.nist.gov/ns/rmf",
                    "value": "b"
                  },
                  {
                    "name": "aggregates",
                    "ns": "http://csrc.nist.gov/ns/rmf",
                    "value": "c"
                  }
                ],
                "label": "organization-defined frequency"
              }  
```

Now, when running `trestle author profile-generate` and `trestle author profile-assemble`, first the generation in the markdown skips the parameter a and don´t show it at all and second in the assemble the original values are not being grabbed from the respective parameters b & c.

Besides that, there´s a need to review why empty values are being shown and in markdown when they shouldn´t be editable at all.

## To Reproduce

1. Click on '....'
1. Scroll down to '....'
1. See error

## Environment

- OS: \[e.g. iOS\]
- Python version:
- Installed packages:

### PR DESCRIPTION
**Pull request by AleJo2995**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

Fixes #1442 

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: tests/trestle/core/commands/author/ssp_test.py [hunk 1]
```diff
@@ -215,7 +215,6 @@ def test_ssp_generate_header_edit(tmp_trestle_dir: pathlib.Path) -> None:
     # confirm new items were added from yaml but not when the same key was alread present (values not updated)
     header, tree = md_api.processor.process_markdown(ac_1)
     assert 'control-origination' in header
-    assert header['x-trestle-set-params']['ac-1_prm_5']['values'] is None
     assert header['x-trestle-set-params']['ac-1_prm_5']['label'] == 'meetings cancelled from cli yaml'
```
Comment (vikas-agarwal76) [inline tests/trestle/core/commands/author/ssp_test.py:218]: @AleJo2995 Why is this assertion removed?

File: trestle/common/const.py [hunk 2]
```diff
@@ -559,6 +559,8 @@
 
 CONTROL_IMPLEMENTATION = 'control-implementation'
 
+AGGREGATES = 'aggregates'
+
 IMPLEMENTED_REQUIREMENT = 'implemented-requirement'
```
Comment (vikas-agarwal76) [inline trestle/common/const.py:564]: There was another parameter property in FedRAMP profile for "alternative". Is that not being handled? For alternative prop also we should just show the alternative parameter name (like aggregates) and. not show the value as the value comes from another (main) parameter.

File: trestle/core/catalog/catalog_writer.py [hunk 3]
```diff
@@ -63,18 +63,6 @@ def write_catalog_as_profile_markdown(
...
-            for param_id, values_dict in control_param_dict.items():
-                props_by_param_ids[param_id] = [
-                    prop for prop in as_list(values_dict.props) if prop.name == 'aggregates'
-                ]
-            to_delete = list({k: v for k, v in props_by_param_ids.items() if v != []}.keys())
-            unique_params_to_del = list(set(to_delete))
-            if unique_params_to_del:
-                delete_list_from_list(control_param_dict, unique_params_to_del)
-
             set_param_dict = self._construct_set_parameters_dict(profile_set_param_dict, control_param_dict, context)
```
Comment (vikas-agarwal76) [inline trestle/core/catalog/catalog_writer.py:66]: Was "aggregates" already handled in current code? Why is this deleted?

File: trestle/core/catalog/catalog_writer.py [hunk 4]
```diff
@@ -174,15 +162,34 @@ def _construct_set_parameters_dict(
...
                 # if the profile doesnt change this param at all, show it in the header with values
                 tmp_dict = ModelUtils.parameter_to_dict(param_dict, True)
                 values = tmp_dict.get('values', None)
-                new_dict = {'id': param_id, 'values': values}
+                # if values are None then don´t display them in the markdown
+                if values is not None:
+                    new_dict = {'id': param_id, 'values': values}
+                else:
+                    new_dict = {'id': param_id, const.PROFILE_VALUES: ['<REPLACE_ME>']}
             new_dict.pop('id', None)
```
Comment (vikas-agarwal76) [inline trestle/core/catalog/catalog_writer.py:178]: profile-values: should be shown in both cases whether a value exists or not for a parameter.

File: trestle/core/commands/author/profile.py [hunk 5]
```diff
@@ -286,30 +286,23 @@ def _replace_modify_set_params(
         return changed
 
     @staticmethod
-    def _add_aggregated_parameter(
-        param: Any, param_dict: Dict[str, Any], control_id: str, controls: Any, param_map: Dict[str, str]
-    ) -> None:
+    def _add_aggregated_parameter(param: Any, param_dict: Dict[str, Any]) -> None:
```
Comment (vikas-agarwal76) [inline trestle/core/commands/author/profile.py:289]: Was aggregated parameter already handled before? if yes was was the code doing for those params.

File: trestle/core/commands/author/profile.py [hunk 6]
```diff
@@ -286,30 +286,23 @@ def _replace_modify_set_params(
...
+        for prop in as_list(agg_props):
+            if param_dict[prop.value].get('values'):
+                agg_param_values = param_dict[prop.value].get('values')
+                param.values = param.values if param.values is not None else []
+                for value in as_list(agg_param_values):
+                    param.values.append(value)
                 dict_param = ModelUtils.parameter_to_dict(param, False)
                 param_dict[param.id] = dict_param
-                param_map[param.id] = control_id
         return None
```
Comment (vikas-agarwal76) [inline trestle/core/commands/author/profile.py:306]: Why is the function signature changes and control_id removed.

File: trestle/core/catalog/catalog_writer.py (hunk 4 above)
Comment (AleJo2995) [reply trestle/core/catalog/catalog_writer.py:178]: Done. Added profile values when values is not None

File: trestle/common/const.py (hunk 2 above)
Comment (AleJo2995) [reply trestle/common/const.py:566]: Done

File: trestle/core/catalog/catalog_reader.py [hunk 7]
```diff
@@ -71,9 +71,15 @@ def read_additional_content(
...
                 alters_map[sort_id] = control_alters
                 for param_id, param_dict in control_param_dict.items():
+                    param_dict[const.VALUES] = param_dict[const.VALUES] if const.VALUES in param_dict else []
                     # if profile_values are present, overwrite values with them
                     if const.PROFILE_VALUES in param_dict:
-                        param_dict[const.VALUES] = param_dict.pop(const.PROFILE_VALUES)
+                        if param_dict[const.PROFILE_VALUES] != [] and param_dict[const.PROFILE_VALUES] is not None:
+                            if not write_mode and '<REPLACE_ME>' in param_dict[const.PROFILE_VALUES]:
+                                param_dict[const.PROFILE_VALUES].remove('<REPLACE_ME>')
+                            param_dict[const.VALUES] = param_dict[const.PROFILE_VALUES]
```
Comment (vikas-agarwal76) [inline trestle/core/catalog/catalog_reader.py:80]: After removing <REPLACE_ME> wouldn't profile-values become empty "" and values should not be set?

File: trestle/core/catalog/catalog_writer.py [hunk 8]
```diff
@@ -174,15 +162,43 @@ def _construct_set_parameters_dict(
...
                 new_dict[const.DISPLAY_NAME] = display_name
-            key_order = (const.LABEL, const.GUIDELINES, const.PROFILE_VALUES, const.VALUES, const.DISPLAY_NAME)
+            key_order = (
+                const.LABEL,
+                const.GUIDELINES,
+                const.VALUES,
+                const.AGGREGATES,
+                const.DISPLAY_NAME,
+                const.ALT_IDENTIFIER,
+                const.PROFILE_VALUES
```
Comment (vikas-agarwal76) [inline trestle/core/catalog/catalog_writer.py:200]: Alt identifier should be above display name. The only 2 editable fields are display name and profile-values and they should be last.

File: trestle/core/catalog/catalog_reader.py [hunk 9]
```diff
@@ -71,9 +71,15 @@ def read_additional_content(
...
+                        if not write_mode:
+                            param_dict.pop(const.PROFILE_VALUES)
                     final_param_dict[param_id] = param_dict
```
[the remark this answers is not in the document]
Comment (AleJo2995) [reply trestle/core/catalog/catalog_reader.py:83]: This is meant to not write replace me or pop profile values in read mode. Meaning that while writing into markdown replace me is not going to be removed or profile values are not as well. Logic before was implemented differently to pop the value out

File: trestle/core/catalog/catalog_reader.py (hunk 7 above)
Comment (AleJo2995) [reply trestle/core/catalog/catalog_reader.py]: Done

**Review by vikas-agarwal76 — APPROVED** [review]
LGTM
